poc/extensions/src/extension_core.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{DispatchError, Dispatchable};
+use crate::{ExtensionId, ExtensionIdTy};
+use parity_scale_codec::{Decode, Encode};
+
+pub trait ExtensionCore: ExtensionId {
+    type Config: Config;
+    fn some_host_function(
+        args: <Self::Config as Config>::ArgsOfSomeHostFunction,
+    ) -> <Self::Config as Config>::ResultOfSomeHostFunction;
+}
+
+pub trait Config {
+    type ArgsOfSomeHostFunction: Decode;
+    type ResultOfSomeHostFunction: Encode;
+}
+
+// #[extension(ExtensionCore)]
+// #[disallow_context(XCM)]
```

Comment:
no need to have this one. it should be implemented in the permission controller and is something to be determined by the runtime, not the extension implementation